From 6574e5383e34755f9e982865f26e64dd87f438c7 Mon Sep 17 00:00:00 2001
From: Gyorgy Sarvari <skandigraun@gmail.com>
Date: Mon, 27 May 2024 18:02:56 +0200
Subject: [PATCH] panfrost add stub methods to report EGL 1.5 conformance

Once the PIPE_CAP_DEVICE_RESET_STATUS_QUERY is missing to
report EGL 1.5 conformance, instead of stucking on 1.4.

Add a stub method, and hope that it doesn't blow up.

Upstream-Status: Inappropriate
---
 src/gallium/drivers/panfrost/pan_context.c | 7 +++++++
 src/gallium/drivers/panfrost/pan_screen.c  | 1 +
 2 files changed, 8 insertions(+)

--- ./src/gallium/drivers/panfrost/pan_context.c.orig	2025-07-16 21:28:34.868817019 +0200
+++ ./src/gallium/drivers/panfrost/pan_context.c	2025-07-16 21:29:47.410388546 +0200
@@ -982,6 +982,12 @@
    *pfence = panfrost_fence_from_fd(pan_context(pctx), fd, type);
 }
 
+static enum pipe_reset_status
+panfrost_get_device_reset_status(struct pipe_context *pipe)
+{
+   return PIPE_NO_RESET;
+}
+
 static void
 panfrost_fence_server_sync(struct pipe_context *pctx,
                            struct pipe_fence_handle *f)
@@ -1031,6 +1037,7 @@
 
    gallium->create_fence_fd = panfrost_create_fence_fd;
    gallium->fence_server_sync = panfrost_fence_server_sync;
+   gallium->get_device_reset_status = panfrost_get_device_reset_status;
 
    gallium->flush = panfrost_flush;
    gallium->clear = panfrost_clear;
